### ISSUE TITLE
LokiFormatter instanciation fails because `labels` defaults to empty string

### ISSUE THREAD

**Issue by victor-keltio** [itspire/monolog-loki#23]
Hey there,
I followed the README to add a Loki with `WhatFailureGroupHandler`.

The README indicates that the config should look like:

```php
'loki' => [
    'formatter_with' => [
        'labels' => env('LOKI_LABELS', ''),
        ...
    ],
    ...
],
```

But when using this config without specifying `LOKI_LABELS`, the logger instanciation fails with:

```
[2024-07-03 09:04:22] laravel.EMERGENCY: Unable to create configured logger. Using emergency logger. {"exception":"[object] (TypeError(code: 0): Itspire
\\MonologLoki\\Formatter\\LokiFormatter::__construct(): Argument #1 ($labels) must be of type array, string given, called in /var/www/my-awesome-app/app/Logging/
LokiNoFailureHandler.php on line 17 at /var/www/my-awesome-app/vendor/itspire/monolog-loki/src/main/php/Formatter/LokiFormatter.php:30)
[stacktrace]
```

I changed the faulty line to:

```php
'labels' => [],
```

But now I can't use a custom `LOKI_LABELS` value.
I don't know much about PHP nor Monolog/Loki, how could we solve this issue? Is it related to my code or is it a bug in the library?

Many thanks :)

**Comment by rrajkomar** [comment]
Hello, 
from your sample code I suspect you're using Laravel.
Althought I'm not familiar with it, from looking at the error message and your example, I'd suspect the issue comes from this line :  
```
'labels' => env('LOKI_LABELS', ''),
```

that should be 
```
'labels' => env('LOKI_LABELS', []),
```

The labels option is an array, not a string.

**Comment by victor-keltio** [comment]
Hey, thanks for that prompt answer, I really appreciate that.

I changed the line to what you suggested, and could get it working with this, but I haven't tried when passing a custom env var.
I think that it won't work as the `env` Laravel function won't cast it automatically to an array. One could use `explode` to convert it to a list, but again, I haven't tried.

I guess the README is outdated then, more specifically [this line](https://github.com/itspire/monolog-loki/blob/master/README.md?plain=1#L172)?

**Comment by rrajkomar** [comment]
Unfortunately, as I mentioned earlier, I'm not familiar with Laravel configs therefore I do not maintain the related documentation myself. It is up to the community to ensure it is correct and up to date.

Feel free to create a PR once you've figured out the proper way to make it work, or maybe contact the people who put the configuration there to see with them how they worked it out.

**Comment by victor-keltio** [comment]
Indeed. Many thanks for your answers. I'll dig a bit further and try to make it work, but as I don't use custom labels I'm worried my work won't be very helpful.

**Comment by victor-keltio** [comment]
I could make it work by using:

```php
                'labels' => env('LOKI_LABELS', '') ? array_reduce(
                    explode('|', env('LOKI_LABELS', '')),
                    function ($carry, $item) {
                        list($key, $value) = explode(',', $item);
                        $carry[$key] = $value;
                        return $carry;
                    },
                    []
                   ) : [],
```

I'll send a PR tomorrow.

**Comment by victor-keltio** [comment]
Fixed by #24
